### DISCUSSION TITLE
How to enforce signature and encryption for receiving messages?

### DISCUSSION THREAD

**Discussion by sopgreg** [phax/phase4#161] [Q&A] [answered]
Is there a way to enforce a (valid) signature and encryption in the receiving AS4 message? Right now, it seems that it's possible to send a completely unencrypted, unsigned message to the `AS4IncomingHandler` and it will process the message without any warnings.

Is there an interface/propery to enforce signatures and encryption?

E.g. here, if the SOAP does not contain a `Security` element, this is just logged in debug mode.

![grafik](https://github.com/phax/phase4/assets/33895520/864c20f5-a6e7-4b48-9eec-93ff4c2ec058)

**Comment by phax** [comment]
I created #162 for that. Makes sense to enforce that.

**Reply by phax** [reply, marked as answer]
As it turns out, the proposed solution to add this to the profile is not sufficient. This would need to be done for every message type separately. Instead the following methods can be used on a case by case basis:
* `IAS4MessageState.isSoapSignatureChecked ()` returns `true` if the message was signed
* `IAS4MessageState.isSoapDecrypted()` returns `true` if the message was decrypted

**Reply by sopgreg** [reply]
This is fine, we will check it that way for our use cases "manually" in our application
